MerlinAU.sh

```diff
@@ -4519,9 +4823,17 @@ _ShowMainMenu_()
    then notificationStr="${REDct}NOT SET${NOct}"
    else notificationStr="${GRNct}$(_SimpleNotificationDate_ "$notifyDate")${NOct}"
    fi
+   isGNUtonFW=$(_GetFirmwareVariantFromRouter_)  
```

Comment:
This call  `isGNUtonFW=$(_GetFirmwareVariantFromRouter_)`  *must* happen outside the "`_ShowMainMenu_`" function and at the earliest point as possible (**independently** of any interactive session) because the global variable "`isGNUtonFW`" must be set and available ***before*** running the "`_RunFirmwareUpdateNow_`" function via CLI, via the Main Menu, or the scheduled cron job.

I suggest the best place to set the "`isGNUtonFW`" global variable would be **after** calling these functions (lines #4594-4595):

```
check_model_support
check_version_support
```

Reply:
@Martinski4GitHub Resolved in commit: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/186/commits/27b062816b8c047f0bf6867865446a98dd27b8d0